GLOSSARY.md

```diff
@@ -0,0 +1,284 @@
+or "unexpected" — unforeseen by the author.
+
+Programmer errors are _always_ unexpected (or else the programmer would not
+have made them!) Operational errors _may be_ unexpected — there may be an
+unaccounted-for operational state that the program enters. Unexpected errors
+are dangerous, and may cause the program to perform undesirable operations. The
+error symposium recommends that programs encountering unexpected errors should
+crash immediately.
+
+∅               | Programmer Error   | Operational Error
+--------------- | ------------------ | ------------------
+**Expected**    | :no_entry_sign:    | :white_check_mark:
+**Unexpected**  | :white_check_mark: | :white_check_mark:
+
+## Post Mortem Analysis
+
+The practice of debugging a program after it has already crashed by inspecting
+the serialized state of the program at the time of error. This usually leverages
+tools like `lldb`, `gdb`, or `mdb` to inspect a "core" (defined below.)
+
+### Abort
+
+The [`abort()`][abort-syscall] system call. Accessible to Node via
+`process.abort()`. Called by V8 when `--abort-on-uncaught-exception` is enabled
+and an exception is thrown without a top-level `catch` handler.
+
+Halts the program and serializes the state of the program (the contents of the
+stack and heap segments, as well as the state of open file descriptors) to a
+"core file." Causes the program to return a non-zero exit status.
+
+### Core / Core dump / Core file
+
+The serialized state of a program, containing the stack and heap segments of
+memory in their entirety. Analyzable via `mdb`, `gdb`, or `lldb`. V8-specific
+plugins are available in the form of `mdb_v8`, `llnode`, and `lldb-v8`, and
+allow users to inspect JavaScript stack frames as well as the frozen object
+graph in memory (including objects pending garbage collection.)
+
+### Stack Frame
+
+A data structure held in the stack segment of memory representing a function
+call. This typically saves the arguments with which the function was called
+as well as a return address to jump to when the call completes.
+
+### Unwinding the Stack / Top of Stack
+
+Stack frames are collected into a stack that grows and shrinks as functions
+are called and functions return. Exceptions work by "unwinding" the stack
+to the nearest exception handler, if any — that is, they remove stack frames
+giving control back to the nearest frame with an exception handler. The
+stack is grown and shrunk in-place, so any subsequent function calls (even
+ones implicitly made by the JS VM) will clobber the original stack leading
+to the error. `Error` stacktraces _record_ some frame information leading to
+the error, but throwing an error is generally a lossy operation.
+
+If no exception handler is registered in a stack, instead of unwinding the
+stack, V8 may abort the program or call a user-defined `OnUncaughtException`
+handler.
+
+## Promise
+
+The 30-second version:
+
+A [well-specified][promises-aplus] pattern that provides a container type
```

Comment:
Related to link:
http://stackoverflow.com/questions/22539815/arent-promises-just-callbacks
http://stackoverflow.com/questions/22519784/how-do-i-convert-an-existing-callback-api-to-promises 
http://bluebirdjs.com/docs/why-promises.html